kore/test/Test/Kore/Step/Step.hs

```diff
@@ -529,17 +529,15 @@ test_applyRewriteRule_ =
             expect :: Either
                 UnificationOrSubstitutionError [OrPattern Variable]
             expect = Right
-                [ OrPattern.fromTermLike
-                    (mkExists
-                        Mock.x
-                        (mkAnd
-                            (mkElemVar Mock.x)
-                            (mkEquals_
-                                (Mock.functional11 (mkElemVar Mock.x))
-                                (Mock.functional10 (mkElemVar Mock.x))
-                            )
-                        )
-                    )
+                [ OrPattern.fromPatterns
```

Comment:
What happened here?

Reply:
Caused by the `ensures` fix. This entire thing is confusing because the ensures clause uses the same `x` as the configuration, and it wasn't being refreshed before this PR.

If we change the configuration to be `y` instead of `x`, then the actual result is the one in this PR (see the first commit in https://github.com/kframework/kore/pull/967 ).

My best guess about what was happening is: the `requires` variable wasn't refreshed, and had the same name as the configuration variable, we detected that we had a rule variable which didn't disappear, so we quantified it. But it should have disappeared, because we matched on the (renamed) x in the rule.